File: Frontend/src/types/article.ts
export type ArticleSource = 'Builtin' | 'Llm'
export type DifficultyLevel = 'Basic' | 'Intermediate' | 'Advanced'
export type CefrLevel = 'A1' | 'A2' | 'B1' | 'B2' | 'C1' | 'C2'
export type RecommendedAction = 'LearnNow' | 'ReviewLater' | 'ChallengeOnly'
export type WordExampleKind = 'contextual' | 'general'

export interface WordExample {
  kind: WordExampleKind
  sentence: string
  explanation: string
}

export interface ArticleSummary {
  id: string
  title: string
  difficultyLevel: DifficultyLevel
  cefrLevel: CefrLevel
  wordCount: number
  source: ArticleSource
  topicTag?: string | null
}

export interface ArticleDetail extends ArticleSummary {
  content: string
  vocabMappings: ArticleVocabMapping[]
}

export interface ArticleVocabMapping {
  id: string
  wordLemma: string
  contextMeaning: string
  phonetics: string
  examples: WordExample[]
  specialUsage: string
  difficultyInContext: DifficultyLevel
  recommendedAction: RecommendedAction
  isKeyVocab: boolean
}

export interface ReadingLog {
  id: string
  articleId: string
  startTime: string
  endTime?: string | null
  durationSeconds: number
  lookupCount: number
  commentsCount: number
}

export interface WordDefinition {
  word: string
  phonetics: string
  meanings: Array<{ definition: string; isContextual: boolean; context: string }>
  collocations: string[]
  examples: WordExample[]
  specialUsage: string
  difficultyLevel: DifficultyLevel
  cefrLevel: CefrLevel
}

export interface ArticleComment {
  id: string
  paragraphIndex: number
  paragraphText: string
  commentText: string
  aiReply?: string | null
  timestamp: string
}

export interface ReadingAgentResponse {
  message: string
  skillCalls: Array<{ skillName: string; summary: string }>
  definition?: WordDefinition | null
  vocabExtract?: {
    keyVocab: Array<{
      word: string
      phonetics: string
      contextMeaning: string
      usageExample?: WordExample | null
      generalExample?: WordExample | null
      difficulty: string
      action: string
    }>
    skippedBasic: string[]
    skippedRare: string[]
  } | null
  commentReply?: { reply: string } | null
}

File: Backend/NextWord.Domain/Entities/Article.cs
using NextWord.Domain.Enums;

namespace NextWord.Domain.Entities;

public sealed class Article
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public string Title { get; set; } = string.Empty;
    public string Content { get; set; } = string.Empty;
    public DifficultyLevel DifficultyLevel { get; set; } = DifficultyLevel.Basic;
    public CefrLevel CefrLevel { get; set; } = CefrLevel.A1;
    public int WordCount { get; set; }
    public ArticleSource Source { get; set; } = ArticleSource.Builtin;
    public Guid? AnnotationId { get; set; }
    public string? TopicTag { get; set; }
    public DateTimeOffset CreatedAt { get; set; } = DateTimeOffset.UtcNow;

    public DifficultyAnnotation? Annotation { get; set; }
    public List<ReadingLog> ReadingLogs { get; set; } = [];
    public List<ArticleComment> Comments { get; set; } = [];
    public List<ArticleVocabMapping> VocabMappings { get; set; } = [];
}
